app/src/main/java/tttTDD/Java/Game.java

```diff
@@ -67,11 +67,21 @@ public boolean performMove(int playerInput) {
         return true;
     }
 
-    private boolean gameHasWinner() {
-        if (board.hasWinningSet(currentPlayer.getMarker())) {
+    private boolean gameHasFinished() {
+        if (isThereAWinner() && !isThereATie()) {
             gameWinner = String.format("Player %s has won", currentPlayer.getMarker());
+        } else {
+            gameWinner ="";
```

Comment:
Empty string here feels a little bit like a magic value (magic string) but I think I'm being overly picky with that statement. -- Completely optional, but consider the idea of using gameWinner = GameStatus.Active (constant/enum of somekind) (don't focus on the naming in the example if you consume it, just the concept in general)